datafusion/core/src/physical_plan/joins/utils.rs

```diff
@@ -324,66 +322,24 @@ pub fn cross_join_equivalence_properties(
 ///
 /// This way; once we normalize an expression according to equivalence properties,
 /// it can thereafter safely be used for ordering equivalence normalization.
-fn get_updated_right_ordering_equivalence_properties(
+fn get_updated_right_ordering_equivalent_class(
     join_type: &JoinType,
-    right_oeq_classes: &[OrderingEquivalentClass],
+    right_oeq_class: &OrderingEquivalentClass,
     left_columns_len: usize,
     join_eq_properties: &EquivalenceProperties,
-) -> Result<Vec<OrderingEquivalentClass>> {
-    let updated_oeqs = match join_type {
+) -> Result<OrderingEquivalentClass> {
+    match join_type {
         // In these modes, indices of the right schema should be offset by
         // the left table size.
         JoinType::Inner | JoinType::Left | JoinType::Full | JoinType::Right => {
-            add_offset_to_ordering_equivalence_classes(
-                right_oeq_classes,
-                left_columns_len,
-            )?
+            let right_oeq_class = right_oeq_class.add_offset(left_columns_len)?;
```

Comment:
I am a big fan of putting the logic into `OrderingEquivalentClass` rather than free functions that take an `OrderingEquivalentClass`